worker/utils/utils.js

```diff
@@ -49,6 +51,27 @@ module.exports = {
     })
   },
 
+  async encryptJob(string1, string2){
+    const secret = this.retrievePassword() + string1 + string2
+    const salt = this.generateSalt();
+    const digest = crypto.scryptSync(secret, salt, 64);
+    return digest.toString('hex');
+  },
+
+  async decryptJob(digest, string1, string2){
+
+    const secret = this.retrievePassword() + string1 + string2;
+    const salt = this.generateSalt();
+    const hash = crypto.scryptSync(secret, salt, 64,);
+    return (hash == digest);
```

Comment:
This probably doesn't matter in this specific case in *practical* terms, but as a general rule, any kind of comparison in cryptographic code should be constant-time using `crypto.timingSafeEqual()`.

(c.f. the OpenBSD implementation of password checking: https://github.com/openbsd/src/blob/b66614995ab119f75167daaa7755b34001836821/lib/libc/crypt/bcrypt.c#L229 )